pkg/ring/kvstore.go

```diff
@@ -1,36 +1,27 @@
 package ring
 
 import (
-	"context"
 	"flag"
 	"fmt"
 	"sync"
 
-	"github.com/golang/protobuf/proto"
-	"github.com/golang/snappy"
+	"github.com/cortexproject/cortex/pkg/ring/kvstore"
+	"github.com/cortexproject/cortex/pkg/ring/kvstore/consul"
+	"github.com/cortexproject/cortex/pkg/ring/kvstore/etcd"
 )
 
 var inmemoryStoreInit sync.Once
-var inmemoryStore KVClient
-
-// KVClient is a high-level client for Consul, that exposes operations
-// such as CAS and Watch which take callbacks.  It also deals with serialisation
-// by having an instance factory passed in to methods and deserialising into that.
-type KVClient interface {
-	CAS(ctx context.Context, key string, f CASCallback) error
-	WatchKey(ctx context.Context, key string, f func(interface{}) bool)
-	WatchPrefix(ctx context.Context, prefix string, f func(string, interface{}) bool)
-	Get(ctx context.Context, key string) (interface{}, error)
-	PutBytes(ctx context.Context, key string, buf []byte) error
-}
+var inmemoryStore kvstore.KVClient
```

Comment:
this doesn't seem to be used anymore, we previously assigned to it in `NewKVStore`

Reply:
Thats a bug; the inmemorystore needs to be a singleton for the single process stuff to work.

Fixed.